### PR TITLE
Move import experiment validation to backend

### PR DESCRIPTION
**Pull request by RidhamShah**
Develop a validation API for experimental data. Now on importation of experimental files, these files will be transmitted to the backend for validation, and the errors associated with each file will be returned. Subsequently, experiments without errors will be imported upon the user's action of clicking the "IMPORT" button.

### CODE REVIEW

**Comment by danoswaltCL** [comment]
@VivekFitkariwala can you review this one?

File: backend/packages/Upgrade/src/api/services/ExperimentService.ts [hunk 1]
```diff
@@ -77,9 +80,21 @@ import { CacheService } from './CacheService';
...
 import { ArchivedStatsRepository } from '../repositories/ArchivedStatsRepository';
+import { validate } from 'class-validator';
+import { plainToClass } from 'class-transformer';
+import { StratificationFactorRepository } from '../repositories/StratificationFactorRepository';
 
 @Service()
 export class ExperimentService {
+  backendVersion = env.app.version;
+  allIdMap = {};
+  errorRemovePart = 'instance of ExperimentDTO has failed the validation:\n - ';
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/src/api/services/ExperimentService.ts:91]: These hardcoded texts can be moved outside of the the class

File: frontend/projects/upgrade/src/app/core/experiments/store/experiments.model.ts [hunk 2]
```diff
@@ -354,3 +354,13 @@ export interface InteractionEffectGraphData {
   series: InteractionEffectLineChartSeriesData[];
   dot: boolean;
 }
+
+export interface ValidateExperimentError {
```
Comment (VivekFitkariwala) [inline frontend/projects/upgrade/src/app/core/experiments/store/experiments.model.ts:358]: Are these types used in other places apart from the component? If not, we should move it inside the component.

**Review by VivekFitkariwala — APPROVED** [review]

File: frontend/projects/upgrade/src/app/core/experiments/store/experiments.model.ts (hunk 2 above)
Comment (RidhamShah) [reply frontend/projects/upgrade/src/app/core/experiments/store/experiments.model.ts]: code updated

File: backend/packages/Upgrade/src/api/services/ExperimentService.ts (hunk 1 above)
Comment (RidhamShah) [reply backend/packages/Upgrade/src/api/services/ExperimentService.ts]: code updated

**Review by VivekFitkariwala — APPROVED** [review]

**Pull request merged**